### PR TITLE
fix: enableEgressStore flag is boolean, not string

### PR DESCRIPTION
**Pull request by maghirardelli**
Issue #, if available: GALI-1071

Description of changes: enableEgressStore flag in the configuration file for SWB is now a boolean and not a string. Serverless accepts this by invoking a script to convert the boolean and read the string value of "true" if it was true and "false" otherwise. This affected the logic of many files that used this flag as well as unit tests of those files. All are updated. Tested with unit tests as well as locally and through the API.

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main/config/settings/.defaults.yml [hunk 1]
```diff
@@ -204,4 +204,8 @@ egressNotificationTopicName: ${self:custom.settings.namespace}-EgressTopic
 # =========================================================================================================
 
 # Determine whether workspace should be accessible only through AppStream
+# NOTE: Once the isAppStremEnabled is set to true, the AppStrem feature will be enabled and can NOT be toggled off.
```
Comment (SanketD92) [inline main/config/settings/.defaults.yml]: AppStream typo (x2)

File: addons/addon-base-raas/packages/base-raas-services/lib/data-source/__tests__/environment-resource-service.test.js [hunk 2]
```diff
@@ -494,6 +494,23 @@ describe('EnvironmentResourceService', () => {
 
   describe('addToKmsKeyPolicy', () => {
     it('add new principal to KMS policy with no principals', async () => {
+      environmentResourceService._settings = {
```
Comment (SanketD92) [inline addons/addon-base-raas/packages/base-raas-services/lib/data-source/__tests__/environment-resource-service.test.js]: Why are these needed now but weren't needed earlier?

**Review by SanketD92 — COMMENTED** [review]
Since the changes touch multiple components, could you list out the manual testing you did in the description (both UI and backend)?

File: main/config/settings/.defaults.yml (hunk 1 above)
Comment (maghirardelli) [reply main/config/settings/.defaults.yml]: Got it--will fix

File: addons/addon-base-raas/packages/base-raas-services/lib/data-source/__tests__/environment-resource-service.test.js (hunk 2 above)
Comment (maghirardelli) [reply addons/addon-base-raas/packages/base-raas-services/lib/data-source/__tests__/environment-resource-service.test.js]: Because there is an environmentResourceService setup in the beforeEach but it didn't include getBoolean so I needed to have the implementation of the mocked getBoolean function. Putting it here overrode the beforeEach one and meant that the environmentResourceService mock this test was using didn't have the methods that the beforeEach one did. Thanks for pointing this out and making me figure this out though--I can just put the getBoolean mock in the beforeEach one which reduces code repetition and still works!

**Comment by maghirardelli** [comment]
> Since the changes touch multiple components, could you list out the manual testing you did in the description (both UI and backend)?

Below are the tests I ran after making the changes everywhere to treat this flag as a boolean instead of a string.

First, I re-ran every unit test suite that used this value. Most of them I had to either change the mocked return value (from `'false'` to `false` or `'true'` to `true`) as well as change the method that was mocked. Before, the mocking of this value was a `get()` mocking and that had to be changed to a `getBoolean()` mocking. All of these unit tests passed after my code changes and testing set-up changes.

Second, I deployed an environment (upgraded from non TRE) with AppStream and Egress Store enabled. I was able to see and click the "Egress Store" button in the UI. Additionally, hitting the `/api/data-egress/:id` API fielded a successful response.

Third, I deployed an environment with AppStream enabled but not Egress Store. After creating a workspace, I confirmed that the "Egress Store" button did not appear in the Workspace details UI page. Additionally, I hit the `/api/data-egress/:id` API and got a not successful response in line with what would be expected if the egress store was not enabled.

Lastly, I took the last environment and turned the Egress Store on and redeployed. After this, the "Egress Store" button appeared on Workspace details UI. For the already made workspace, clicking this button did nothing but give an error (which is to be expected). After adding a new configuration for a workspace type, I created a new workspace with this new configuration. It's detail page displayed the "Egress Store" button and clicking it displayed the table correctly. Additionally, I hit the `/api/data-egress/:id` API with the new workspace id and got a success response.

**Review by SanketD92 — APPROVED** [review]
LGTM
